yt_idefix/data_structures.py

```diff
@@ -361,6 +395,8 @@ def _parse_header_file(self):
         # definitions.h is presumed to be along with data file
         if self._definitions_header is None:
             self._definitions_header = os.path.join(self.fullpath, "definitions.h")
+        elif not os.path.isfile(self._definitions_header):
+            raise FileNotFoundError(f"No such file {self._definitions_header!r}")
```

Comment:
It's not, but it's not very well known. This is a way to format the string using the repr method, so that strings are guaranteed to be correctly quoted, and we call `__repr__` instead of `__str__` on other objects.
```
>>> file = "hello.png"
>>> print(f"No such file {file!r}")
No such file 'hello.png'
>>> print(f"No such file {file}")
No such file hello.png
```

Reply:
Hah now I'm sure it's a typo.
```suggestion
            raise FileNotFoundError(f"No such file {self._definitions_header}!")
```

Reply:
Thanks for your explaination ! I got it !